src/app/static/src/app/components/plots/table/Table.vue

```diff
@@ -0,0 +1,183 @@
+    filteredDataSize: any[],
+    combinedData: any[],
+    flattenedAreas: Dict<NestedFilterOption>,
+    selectedAreaIds: string[],
+    selectedAreaFilterOptions: FilterOption[],
+}
+const props = {
+    tabledata: {
+        type: Array
+    },
+    filters: {
+        type: Array
+    },
+    indicators: {
+            type: Array
+        },
+    indicatorsSize: {
+        type: Array
+    },
+    areaFilterId: {
+        type: String
+    },
+    selectedFilterOptions: {
+        type: Object
+    },
+    selections: {
+        type: Object
+    }
+}
+export default Vue.extend<{}, {}, Computed, Props>({
+    name: "table-view",
+    props: props,
+    data() {
+      return {
+          styleObject: {
+            width: '150px'
+          }
+      }
+    },
+    computed: {
+        nonAreaFilters() {
+             return this.filters.filter((f: Filter) => f.id != this.areaFilterId);
+         },
+        areaFilter() {
+                 return this.filters.find((f: Filter) => f.id == this.areaFilterId)!!;
+             },
+        flattenedAreas() {
+                 return this.areaFilter ? flattenOptions(this.areaFilter.options) : {};
+             },
+        selectedAreaIds() {
+            const selectedAreaIdSet = flattenToIdSet(this.selectedAreaFilterOptions.map(o => o.id), this.flattenedAreas);
+            const areaArray = Array.from(selectedAreaIdSet)
+            if (this.selections.detail === 0 || !this.selections.detail){
+                return [areaArray[0]]
+            } else return areaArray.filter(val => parseInt(val[4]) === this.selections.detail);
+
+        },
+        selectedAreaFilterOptions() {
+            const selectedOptions = this.selections.selectedFilterOptions[this.areaFilterId];
+            if (selectedOptions && selectedOptions.length > 0) {
+                return selectedOptions
+            }
+            return this.areaFilter ? this.areaFilter.options : []; //consider all top level areas to be selected if none are
+        },
+        filteredData() {
+            const result: any[] = [];
+            iterateDataValues(this.tabledata,
+                this.indicators,
+                this.selectedAreaIds,
+                this.nonAreaFilters,
+                this.selections.selectedFilterOptions,
+                (areaId: string, indicatorMeta: ChoroplethIndicatorMetadata, value: number, row: object) => {
+                    result.push({areaId, indicatorMeta, value, row});
+                });
+
+            if (result.length > 0){
+
+                let filterObject: any = {...this.selections.selectedFilterOptions}
+                delete filterObject['area']
+                Object.keys(filterObject).map(function(key, index) {
+                    if (filterObject[key].length < 1 || key === 'area') {
+                        delete filterObject[key]
+                    } else if (key === 'survey'){
+                        filterObject[key] = filterObject[key][0]['label']
```

Comment:
This won't work when you've got multiple values for a filter as in the barchart I think. I'd be inclined not to worry about the filter labels from the seleted filters object until after you've constructed the complete set of rows based on values. It feels safer to me to compare values and just apply labels when you're ready to display.